projects/fourier/lib/fourier/utilities/system.rb

```diff
@@ -5,7 +5,7 @@ module Fourier
   module Utilities
     module System
       def self.system(*args)
-        Kernel.system(*args) || Kernel.abort
```

Comment:
I removed the `Kernel.abort` portion here to allow the exit status of the `system()` call to be returned so it could be handled. Will have unintended side-effects?

Reply:
@pepibumur thoughts on this? Generally, I think it should be fine but I must admit it's more of an educated guess 😐

Reply:
`Kernel.system` returns a boolean to indicate the success of the execution. Therefore the assumption we'll be returning the exit code is not correct. Because we are not expecting automation to be built upon this, I think exiting with 1 should be sufficient. If this changes in the future, I'd start by defining how to use exit codes. Surfacing codes from underlying commands is not good either because they need to be mapped into Fourier's domain and documented.

Reply:
OK, do you have any other suggestions on how I might detect if the command returned a failure? If I add the `Kernel.abort` bit back in, the process just ends there, so I'm not sure how to know within [`format.rb`](https://github.com/tuist/tuist/pull/3451/files#diff-947c174a3251b50f8ce3128a79d8cb3e0b4b6c71bd65b8a46874b72b38c999b2R8) whether or not the linting failed.

Reply:
If the suggestion in this [thread](https://stackoverflow.com/questions/18728069/ruby-system-command-check-exit-code) is accurate. You should be able to do:

```rb
system("command")
$?.exitstatus
```

Reply:
So I've tried the suggestion like so:

```ruby
Kernel.system(*args) || Kernel.abort
$?.exitstatus
```

If the `Kernel.system(*args)` call fails, it calls `Kernel.abort` which returns out of the function, meaning the `$?.exitstatus` is not returned.

@pepibumur I see you changed your review status to approved. Have you changed your mind on if this code needs to be reverted?